### PR TITLE
Combination and product of features for RST

### PR DESCRIPTION
**Pull request by moreymat**
This PR introduces a not-very-elegant, but currently acceptable, way of computing features that are the combination or the product of features.
Small-scale tests show these changes should not make any (significant) difference in performance on RST.

### CODE REVIEW

**Comment by moreymat** [comment]
I would like to refactor feature extraction completely into something like `sklearn.pipeline.FeatureUnion`: http://scikit-learn.org/stable/modules/pipeline.html#featureunion-composite-feature-spaces
http://scikit-learn.org/stable/auto_examples/hetero_feature_union.html

My attempts so far at really bridging the gap between our current setup and a proper pipeline using feature unions have failed because the differences are too important.
We will get there, eventually.

File: educe/rst_dt/learning/doc_vectorizer.py [hunk 1]
```diff
@@ -217,28 +221,36 @@ def _extract_feature_vectors(self, doc):
...
+                                               feat_dict['EDU2'],
+                                               feat_dict['pair']))
+            # combine features
+            feat_dict['pair'].update(feat_comb(feat_dict['EDU1'],
+                                               feat_dict['EDU2'],
+                                               feat_dict['pair']))
+            # add suffix to single EDU features
+            feat_dict['EDU1'] = dict(re_emit(feat_dict['EDU1'].items(), '_EDU1'))
+            feat_dict['EDU2'] = dict(re_emit(feat_dict['EDU2'].items(), '_EDU2'))
+            # convert to list
```
Comment (kowey) [inline educe/rst_dt/learning/doc_vectorizer.py]: Oh right, that explains why my just-point-to-the-dict suggestion won't work.  You need a copy if you want to rename, right…

**Comment by kowey** [comment]
OK well, I didn't really look much into this.  Merging now.  Want me to run these?

**Comment by moreymat** [comment]
In theory, it should work just as the previous versions.
I did not test it as thoroughly as usual, so will check this on Monday.

2015-04-26 12:47 GMT+02:00 Eric Kow <notifications@github.com>:

**Pull request merged**